### PR TITLE
Sync LHLS progressive with upstream v1.0.0 PR

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...
Pull in upstream fmp4, aac and mp3 fixes https://github.com/video-dev/hls.js/pull/2370

### Why is this Pull Request needed?
For functional tests to pass. Mp3 audio and fmp4 playback test failed in these tests and in the player.

### CODE REVIEW

File: src/utils/mp4-tools.ts [hunk 1]
```diff
@@ -331,6 +331,13 @@ export function getDuration (data, initData) {
     const scale = initData[id].timescale || 90e3;
     duration += rawDuration / scale;
   }
+  if (duration === 0) {
+    // If duration samples are not available in the traf use sidx subsegment_duration
+    const sidx = parseSegmentIndex(data);
+    if (sidx && sidx.references) {
+      return sidx.references.reduce((dur, ref) => dur + ref.info.duration || 0, 0);
```
Comment (jnatalzia) [inline src/utils/mp4-tools.ts]: this `||` is rescuing for `NaN` I assume?

Comment (robwalch) [reply src/utils/mp4-tools.ts]: or `undefined`

File: src/demux/mp3demuxer.ts [hunk 2]
```diff
@@ -9,13 +9,20 @@ import { dummyTrack } from './dummy-demuxed-track';
...
+  private config: any;
   private _audioTrack!: any;
   private _id3Track!: DemuxedTrack;
-  private cachedData: Uint8Array = new Uint8Array();
   private frameIndex: number = 0;
+  private cachedData: Uint8Array = new Uint8Array();
   private initPTS: number | null = null;
   static readonly minProbeByteLength: number = 4;
 
+  constructor (observer, config) {
```
Comment (zetagame) [inline src/demux/mp3demuxer.ts]: Not necessary

Comment (robwalch) [reply src/demux/mp3demuxer.ts]: Updated. Now aac and mp3 demuxers are base classed to remove duplicate code.

**Pull request merged**